File: OSCommon/org/optimizationservices/oscommon/exception/OSException.cs
using System;

namespace org.optimizationservices.oscommon.exception{
	/// <summary>
	/// The <c>OSException</c> class is the the parent class of 
	/// all the OS related exception classes. 
	/// @author Jun Ma	
	/// @version 1.0, 09/01/2005
	/// @since OS 1.0
	/// @copyright (c) 2005
	/// </summary>
	public abstract class OSException : System.ApplicationException{

		/// <summary>
		/// Default constructor. 
		/// </summary>
		public OSException(){
		}//constructor

		/// <summary>
		/// Constructs a new exception with the specified detail message.  
		/// </summary>
		/// <param name="message">holds the detail message</param>
		public OSException(string message) : base(message){
		}//constructor
    
		/// <summary>
		/// get exception id
		/// </summary>
		/// <returns> get exception id</returns>
		public abstract int getExceptionId();
	
		/// <summary>
		/// get exception name
		/// </summary>
		/// <returns> get exception name</returns>
		public abstract string getExceptionName();
	
		/// <summary>
		/// get exception type
		/// </summary>
		/// <returns> get exception type</returns>
		public abstract int getExceptionType();

	}//class OSException
}//namespace

File: OSCommon/src/org/optimizationservices/oscommon/exception/OSException.java
/**
 * @(#)OSException 1.0 03/14/2004
 *
 * Copyright (c) 2004
 */
package org.optimizationservices.oscommon.exception;

/**
 *
 * <P>The <code>OSException</code> class is the the parent class of 
 * all the Optimization Services related exception classes. 
 *  
 * </p>
 * @author Robert Fourer, Jun Ma, Kipp Martin
 * @version 1.0, 03/14/2004
 * @since OS 1.0
*/
public abstract class OSException extends Exception {
	/**
	 *
	 * Default constructor. 
	 */
	public OSException(){
	}//constructor
	
    /**
     * Constructs a new exception with the specified detail message.  
     *
     * @param message holds the detail message.
     */
    public OSException(String message) {
    	super(message);
    }//constructor
    
	/**
	 * get exeption id
	 * 
	 * @return exception id
	 */
	public abstract int getExceptionId();
	
	/**
	 * get exception name
	 * 
	 * @return exception name
	 */
	public abstract String getExceptionName();
	
	/**
	 * get exception type
	 * 
	 * @return exception type
	 */
	public abstract int getExceptionType();
	
	/**
	 * main for test purposes.
	 *
	 * </p>
	 *
	 * @param argv command line arguments.
	 */
	public static void main(String[] args){
	}//main
}//class OSException
